### ISSUE TITLE
popups like LSP signature help change Nvim window title

### ISSUE THREAD

**Issue by charbelnicolas** [neovim/neovim#19380]
### Neovim version (nvim -v)

NVIM v0.7.2

### Vim (not Nvim) behaves the same?

DOES NOT APPLY?

### Operating system/version

Arch Linux

### Terminal name/version

Kitty

### $TERM environment variable

xterm-kitty

### Installation

Arch User Repository AUR

### How to reproduce the issue

It's easy to reproduce when using a buffer with an LSP attached, just trigger `vim.lsp.buf.signature_help` or `vim.diagnostic.open_float` and click on the pop up and the neovim window title will change to "_[No Name] - - NVIM_".

### Expected behavior

I would expect the window title to stay the same since I'm actually still inside that file and just doing some basic word processing? I can understand when telescope pops up or fzf but it gets annoying when using completion plugins which can lead to issues like this one:

[nvim-cmp issue](https://github.com/hrsh7th/nvim-cmp/issues/309)

### Actual behavior

Neovim's window title flickers when going through menu entries using completion plugins like nvim-cmp.

[nvim-cmp issue](https://github.com/hrsh7th/nvim-cmp/issues/309)

**Comment by justinmk** [comment]
I guess such popups could temporarily unset 'title', then restore it on WinLeave. `nvim_open_win()` also has a `style=minimal` option, maybe this could fit there (though it seems to target window-local options mostly).

**Comment by Shougo** [comment]
> I guess such popups could temporarily unset 'title', then restore it on WinLeave. nvim_open_win() also has a style=minimal option, maybe this could fit there (though it seems to target window-local options mostly).

The options is not restored.

```
vim.opt.title = true
vim.opt.titlestring = "%F %r"
```

The title is changed when window move.
`nvim-cmp` seems move windows to update the popup window.
So it flickers.

**Comment by justinmk** [comment]
my suggestion is to restore it, or make it part of nvim_open_win functionality.

**Comment by aymanbagabas** [comment]
I'm facing the same issue here too. Popup windows set the title and that causes title change flickering. Super annoying. IMO, popup windows shouldn't be able to set the title.

**Comment by UtkarshVerma** [comment]
I'm also affected by this. I hope it gets implemented.

**Comment by justinmk** [comment]
Proposal from https://github.com/neovim/neovim/pull/22457#issuecomment-1448655680 :

> Often float popups are also used for more long lived things, like terminal sessions. Also, a split window can also be a temporary input window. If we add a feature to disable title setting, it should be (both ways) independent of `w_floating` .

**Comment by glepnir** [comment]
I can't reproduce this **flickers** on master with this minimal config

```vim
set runtimepath+=~/workspace/repro/nvim-cmp
set runtimepath+=~/workspace/repro/cmp-nvim-lsp-signature-help
set runtimepath+=~/workspace/repro/cmp-nvim-lsp
set runtimepath+=~/workspace/nvim-lspconfig

set completeopt=menu,menuone,noinsert

set title
let &titlestring="%F %r"

lua <<EOF
require('cmp_nvim_lsp').setup()
  local cmp = require'cmp'

  -- Set up lspconfig.
  local capabilities = require('cmp_nvim_lsp').default_capabilities()
  require('lspconfig')['lua_ls'].setup {
    capabilities = capabilities
  }

  cmp.setup({
    mapping = cmp.mapping.preset.insert({
      ['<C-b>'] = cmp.mapping.scroll_docs(-4),
      ['<C-f>'] = cmp.mapping.scroll_docs(4),
      ['<C-Space>'] = cmp.mapping.complete(),
      ['<C-e>'] = cmp.mapping.abort(),
      ['<CR>'] = cmp.mapping.confirm({ select = true }), -- Accept currently selected item. Set `select` to `false` to only confirm explicitly selected items.
    }),
    sources = cmp.config.sources({
      {name = 'nvim_lsp', keyword_length = 1},
      { name = 'nvim_lsp_signature_help' }
    })
  })

EOF
```

**Comment by justinmk** [comment]
May have been fixed by `:help focusable` behavior changes? @glepnir in the floatwin what are the values of `hide` and `focusable` reported by `nvim_win_get_config()` ?

**Comment by glepnir** [comment]
type ? they are boolean.   previous fix for window focusable probably also fixed this issue, as the flickering here seems to be caused by win_enter -> maketitle. Maybe we can close this issue now.

**Comment by justinmk** [comment]
yeah, i'm asking if the floatwin has `focusable=false` and/or `hide=true`, which might have fixed the issue

**Comment by glepnir** [comment]
I tried version from master to 0.10, and my minimal config couldn't reproduce the flickering. Let's add a "need-response" label...

**Comment by justinmk** [comment]
but just for reference, in the floating window in your [repro steps](https://github.com/neovim/neovim/issues/19380#issuecomment-3065023020), can you check the  value of `focusable` and `hide`, using `nvim_win_get_config()` in the window

**Comment by glepnir** [comment]
default value  `true false` cmp does not set them
